### PR TITLE
cookbook - multiclass_linearmachine

### PR DESCRIPTION
**Pull request by OXPHOS**

### CODE REVIEW

File: doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Imagine we have files with training and test data. We create CDenseFeatures (here 64 bit floats aka RealFeatures) and :sgclass:`CMulticlassLabels` as
+
+.. sgexample:: multiclass_linearmachine.sg:create_features
+
+We use :sgclass:`CLibLinear` as base classifier. We create an instance of :sgclass:`CLibLinear` with L2 regularized classification. Then we set the residual convergence parameter and enable the bias.
+
+.. sgexample:: multiclass_linearmachine.sg:create_classifier
+
+In order to run :sgclass:`CLinearMulticlassMachine`, we need to specify a multi-class strategy.
```
Comment (lisitsyn) [inline doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst]: What strategy did we choose and why?

File: doc/cookbook/source/conf.py [hunk 2]
```diff
@@ -303,6 +303,7 @@
 
 # options for external link plugin
 extlinks = {
+'cookbook_classifier': ('http://shogun.ml/cookbook/latest/examples/classifier/%s.html',        ''),
```
Comment (karlnapf) [inline doc/cookbook/source/conf.py]: I like this idea, but please use relative links? I.e. remove the `http://shogun.ml/cookbook/latest/`

File: doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
+==========================
+Multi-class Linear Machine
+==========================
+
+We expand the application of linear machines to multi-class dataset by constructing generic multiclass classifiers with ensembles of binary classifiers.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst]: We "extend" ..... 
"datasets" (plural)

File: doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+In this example, we show how to apply :sgclass:`CLibLinear` to multi-class cases with :sgclass:`CLinearMulticlassMachine`.
+
+See :cookbook_classifier:`linear_svm` for the infomration about :sgclass:`CLibLinear` binary classifier.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst]: love this, but the "linear_svm" name is not the nicest, can you just alias it as "see the linear SVM cookbook", where "linear SVM" is the link?

File: doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+-------
+Example
+-------
+
+Imagine we have files with training and test data. We create CDenseFeatures (here 64 bit floats aka RealFeatures) and :sgclass:`CMulticlassLabels` as
+
+.. sgexample:: multiclass_linearmachine.sg:create_features
+
+We use :sgclass:`CLibLinear` as base classifier. We create an instance of :sgclass:`CLibLinear` with L2 regularized classification. Then we set the residual convergence parameter and enable the bias.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst]: Just leave these parameters set to default values (this is already covered in the other cookbook), and don't talk about them here

**Comment by karlnapf** [comment]
Great! Thanks.
Some minor things, and it needs integration testing data

File: doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst (hunk 4 above)
Comment (OXPHOS) [inline doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst]: I can only think of doing it case by case in conf.py..
like :cookbook_linear_svm: `*` redirects to a fixed page

**Comment by OXPHOS** [comment]
@karlnapf should be ready

**Comment by karlnapf** [comment]
https://travis-ci.org/shogun-toolbox/shogun/jobs/135660349#L3989

There is no integration test for this executed, which is because you forgot to update ´data´ to your latest data commit

File: doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+Imagine we have files with training and test data. We create CDenseFeatures (here 64 bit floats aka RealFeatures) and :sgclass:`CMulticlassLabels` as
+
+.. sgexample:: multiclass_linearmachine.sg:create_features
+
+We use :sgclass:`CLibLinear` as base classifier and create an instance of :sgclass:`CLibLinear`.
+
+.. sgexample:: multiclass_linearmachine.sg:create_classifier
+
+In order to run :sgclass:`CLinearMulticlassMachine`, we choose to use one-vs-one reduction strategy.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/classifier/multiclass_linearmachine.rst]: Can you maybe mention the other possible strategy here (using sgclass) ?

**Comment by OXPHOS** [comment]
@karlnapf ready - it is failing the gmm. gmm integration test data was pushed earlier but the meta example is not ready yet

**Comment by karlnapf** [comment]
merging as travis error is unrelated
Thanks!

**Pull request merged**